src/IrcConnection/IrcConnection.cs

```diff
@@ -1050,17 +1050,65 @@ private void _OnConnectionError(object sender, EventArgs e)
             } catch (ConnectionException) {
             }
         }
-        
+
+        /// <summary>
+        /// 
+        /// </summary>
+        private abstract class FiniteThread
```

Comment:
Git blame doesn't reveal much:

    * IrcConnection.cs: Fixed IrcConnection.Disconnect(), the read and
      write thread might thought it needs to reconnect!

It probably was a race condition, not sure between which code exactly...